### DISCUSSION TITLE
Nonconvergence of radiation RTE equation in coaxial flame and setting of radiation boundary conditions

### DISCUSSION THREAD

**Discussion by grabtheNS** [AMReX-Combustion/PelePhysics#548] [Q&A]
Hello, everyone.
I encountered an MLMG solver solving more than the number of max iterations [1000 Max iterations] (also encountered divergence cases)
I'm sure the problem comes from solving the radiation RTE equation, i have try to https://amrex-combustion.github.io/PeleLMeX/manual/html/Troubleshooting.html ，and reduce the time step (as low as 2E-6)But it doesn't seem to work. my case is similiar to Exec/Production/DiffBunsen2D 
```

peleLM.lo_bc = Symmetry      Inflow
peleLM.hi_bc = SlipWallAdiab Outflow
```
I suspect that it is the problem of boundary conditions. I have consulted the code of the radiation part, and it seems that only the fixed boundary conditions of robin type (fixed parameters) can be selected for radiation boundary conditions. This may conflict with my model boundary conditions (which I have symmetric boundary conditions on one side),
https://github.com/AMReX-Combustion/PelePhysics/blob/development/Source/Radiation/PeleLMRad.H row266
```
      if (robin_cell) {
        robin_a_fab(i, j, k) = -1.0 / betafab(i, j, k);
        robin_b_fab(i, j, k) = -2.0 / 3.0;
        robin_f_fab(i, j, k) = 0.0;
      }
```
But I can't locate the exact source of the problem, before the radiation calculation, my flame (with soot) has achieved steady state, if someone has a similar problem, we can communicate and solve it

**Comment by baperry2** [comment]
@wjge do you have any suggestions on what might be happening here or how the boundary conditions should be set up in this case?

**Reply by grabtheNS** [reply]
After my test I found that it is indeed a boundary condition problem, I modified the radiation boundary part of the code, keep the inlet as a constant blackbody radiation, the exit 0 gradient condition, the wall took the original code boundary condition, finally it got the result to my satisfaction!

**Reply by wjge** [reply]
You should be able to specify Dirichlet or Neumann boundary type through pelerad.lo_bc, pelerad.hi_bc in the input file, not limited by Robin type. We will add the wall emission to the P1 approximation in the future updates.
